compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageContext.java

```diff
@@ -212,6 +214,19 @@ ModuleCompilation getModuleCompilation(ModuleContext moduleContext) {
     PackageCompilation getPackageCompilation() {
         if (packageCompilation == null) {
             packageCompilation = PackageCompilation.from(this, this.compilationOptions());
+            
+            // Check if the compiler crashed
+            Collection<Diagnostic> errors = packageCompilation.diagnosticResult().errors();
+            if (errors != null) {
```

Comment:
Is this null check required?

Reply:
Yes, the diagnostics in `PackageManifest` are added without a code. 
https://github.com/ballerina-platform/ballerina-lang/blob/4175b5491ca992402a9ff93a47db3899ad7acb31/compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java#L438

Will create a separate issue to address that.

Reply:
Yes, the diagnostics in `PackageManifest` are added without a code. 
https://github.com/ballerina-platform/ballerina-lang/blob/4175b5491ca992402a9ff93a47db3899ad7acb31/compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java#L438

Will create a separate issue to address that.

Reply:
We can remove it.